### PR TITLE
NODE-2412 Fix handle of modifying last block by blockchain updates

### PR DESCRIPTION
**Pull request by xrtm000**

### CODE REVIEW

File: grpc-server/src/main/scala/com/wavesplatform/events/Repo.scala [hunk 1]
```diff
@@ -65,12 +65,17 @@ class Repo(dbDirectory: String, blocksApi: CommonBlocksApi)(implicit s: Schedule
...
-
     val ba = BlockAppended.from(block, diff, blockchainBeforeWithMinerReward)
-    liquidState = Some(LiquidState(ba, Seq.empty))
+
+    liquidState.fold {
+      liquidState = Some(LiquidState(None, ba, Seq.empty))
+    } {
+      ls =>
+        val solidified = ls.solidify()
+        db.put(keyForHeight(ls.keyBlock.height), solidified.protobuf.update(_.append.update(_.block.modify(_.copy(block = None)))).toByteArray)
```
Comment (Karasiq) [inline grpc-server/src/main/scala/com/wavesplatform/events/Repo.scala]:
```suggestion
        db.put(keyForHeight(ls.keyBlock.height), solidified.protobuf.update(_.append.block.optionalBlock := None)).toByteArray)
```

Comment (xrtm000) [reply grpc-server/src/main/scala/com/wavesplatform/events/Repo.scala]: Done

File: grpc-server/src/test/scala/com/wavesplatform/events/BlockchainUpdatesSpec.scala [hunk 2]
```diff
@@ -536,11 +540,89 @@ class BlockchainUpdatesSpec extends FreeSpec with WithDomain with ScalaFutures w
...
+    "should handle modifying last block correctly" in {
+      @volatile var suspend   = true
+      @volatile var startRead = false
+      withDomainAndRepo(
+        { (d, repo) =>
+          (1 to 5).foreach(_ => d.appendBlock())
+
+          val subscription = Future {
+            val s = repo.createSubscription(SubscribeRequest.of(1, 5))
+            while (repo.height == 5) {}
```
Comment (Karasiq) [inline grpc-server/src/test/scala/com/wavesplatform/events/BlockchainUpdatesSpec.scala]:
```suggestion
            while (repo.height == 5) Thread.sleep(10)
```

File: grpc-server/src/test/scala/com/wavesplatform/events/BlockchainUpdatesSpec.scala [hunk 3]
```diff
@@ -536,11 +540,89 @@ class BlockchainUpdatesSpec extends FreeSpec with WithDomain with ScalaFutures w
...
+            s.cancel()
+            s
+          }.flatten
+
+          val modifyBlock = Future {
+            while (!startRead) {}
```
Comment (Karasiq) [inline grpc-server/src/test/scala/com/wavesplatform/events/BlockchainUpdatesSpec.scala]:
```suggestion
            while (!startRead) Thread.sleep(10)
```

File: grpc-server/src/test/scala/com/wavesplatform/events/BlockchainUpdatesSpec.scala [hunk 4]
```diff
@@ -536,11 +540,89 @@ class BlockchainUpdatesSpec extends FreeSpec with WithDomain with ScalaFutures w
...
+
+    override def iterator(options: ReadOptions): DBIterator = new DBIterator {
+      private val iterator = db.iterator()
+
+      override def next(): Map.Entry[Array[Byte], Array[Byte]] = iterator.next()
+      override def close(): Unit                               = iterator.close()
+      override def seek(key: Array[Byte]): Unit                = iterator.seek(key)
+      override def hasNext: Boolean = {
+        onHasNext()
+        while (suspend()) {}
```
Comment (Karasiq) [inline grpc-server/src/test/scala/com/wavesplatform/events/BlockchainUpdatesSpec.scala]:
```suggestion
        while (suspend()) Thread.sleep(10)
```

**Review by Karasiq — CHANGES_REQUESTED** [review]

**Review by Karasiq — APPROVED** [review]

**Review by DrBlast — APPROVED** [review]

**Review by phearnot — APPROVED** [review]

**Pull request merged**